homekit/StateManager.js

```diff
@@ -360,246 +441,255 @@ module.exports = (device, platform) => {
 
 		set: {
 			ACActive: (state, callback) => {
-				const status = !!state
+				state = !!state
+				log.easyDebug(device.name, '(SET) - AC Active State:', state)
 
-				log.easyDebug(device.name + ' -> Setting AC state Active:', status)
-
-				if (status) {
+				if (state) {
 					device.state.active = true
 					const lastMode = device.HeaterCoolerService.getCharacteristic(Characteristic.TargetHeaterCoolerState).value
 					const mode = characteristicToMode(lastMode)
-
-					log.easyDebug(device.name + ' -> Setting Mode to', mode)
+					log.easyDebug(device.name, '(SET) - HeaterCooler State:', mode)
 					device.state.mode = mode
 				} else if (device.state.mode === 'COOL' || device.state.mode === 'HEAT' || device.state.mode === 'AUTO') {
 					device.state.active = false
 				}
 
+				updateClimateReact()
```

Comment:
Definitely. Because this is such a painful merge, and it's yet to have been tested even, I'm trying to not add complexity for now. We can do this after the merge, once we see everything works. At the end of the day, this simply updates the internal state within the plugin, if the enableClimateReactAutoSetup is False, this does not get POSTed to Sensibo's API so I think there's very little impact here doing this.